apps/playground/src/app/quiz/containers/question/question.component.ts

```diff
@@ -0,0 +1,256 @@
+import { Component, OnInit, Input, Output } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { FormGroup } from '@angular/forms';
+
+import { QuizQuestion } from '../../model/QuizQuestion';
+
+@Component({
+  selector: 'codelab-question-container',
+  templateUrl: './question.component.html',
+  styleUrls: ['./question.component.scss']
+})
+export class QuestionComponent implements OnInit {
+  @Input() answer: string;
+  @Input() formGroup: FormGroup;
+  @Output() question: QuizQuestion;
+  totalQuestions: number;
+  completionTime: number;
+  correctAnswersCount = 0;
+
+  questionID = 0;
+  currentQuestion = 0;
+  questionIndex: number;
+  correctAnswer: boolean;
+  hasAnswer: boolean;
+  disabled: boolean;
+  quizIsOver: boolean;
+  progressValue: number;
+  timeLeft: number;
+  timePerQuestion = 20;
+  interval: any;
+  elapsedTime: number;
+  elapsedTimes = [];
+  blueBorder = '2px solid #007aff';
+
+  allQuestions: QuizQuestion[] = [
+    {
+      questionId: 1,
+      questionText: 'What is the objective of dependency injection?',
+      options: [
+        { optionValue: '1', optionText: 'Pass the service to the client.' },
+        { optionValue: '2', optionText: 'Allow the client to find service.' },
+        { optionValue: '3', optionText: 'Allow the client to build service.' },
+        { optionValue: '4', optionText: 'Give the client part service.' }
+      ],
+      answer: '1',
+      explanation: 'a service gets passed to the client during DI',
+      selectedOption: ''

```

Comment:
Why is this needed?

Reply:
ok I'll create an array to store the selectedOption for each question

Reply:
I'm saving the selected options for the accordion in results; accordion needs BrowserAnimationsModule and the quiz app is using CommonModule.

Reply:
I've changed to use mat-accordion instead of the other implementation.
mat-accordion doesn't seem to be working with *ngFor for some reason, I put in dummy data and it works fine.

Reply:
Hey Kirill, I am using the selected field to check if the field has been selected or not in the dummy QuestionComponent. I am currently using selected?: boolean in the Option.model.ts file and setting option -> option.selected = false or true. Is this good practice? I wasn't sure if you wanted me to infer the selected variable like with questionId/questionIndex... Please let me know. Thanks.